sdcm/nemesis.py

```diff
@@ -1830,14 +1830,24 @@ def repair_nodetool_rebuild(self):
             self.target_node.run_nodetool('rebuild', retry=0)
 
     def disrupt_nodetool_cleanup(self):
-        # This fix important when just user profile is run in the test and "keyspace1" doesn't exist.
-        test_keyspaces = self.cluster.get_test_keyspaces()
-        for node in self.cluster.nodes:
+        # Inner disrupt function for ParallelObject
+        def _dis_nodetool_cleanup(node):
+            # This fix important when just user profile is run in the test and "keyspace1" doesn't exist.
+            test_keyspaces = self.cluster.get_test_keyspaces()
             InfoEvent('NodetoolCleanupMonkey %s' % node).publish()
             with adaptive_timeout(Operations.CLEANUP, node, timeout=HOUR_IN_SEC * 48):
                 for keyspace in test_keyspaces:
                     node.run_nodetool(sub_cmd="cleanup", args=keyspace)
 
+        results = ParallelObject(self.cluster.nodes, num_workers=min(32, len(self.cluster.nodes))).\
+            run(_dis_nodetool_cleanup, ignore_exceptions=True, unpack_objects=True)
+        runs_that_finished_with_exception = [res for res in results if res.exc]
+        if runs_that_finished_with_exception:
+            self.log.error("Nodetool cleanup failed on one of the nodes",
```

Comment:
The number of failures is unknown. So, need to rephrase it.

Reply:
Even though we don't have number of failures the error message will include all the exceptions.
If you say its important to print the exact number of failures in this text message we should not use generator in previous code block. The length of list 'runs_that_finished_with_exception' gives us exactly what you ask for.

Reply:
I don't ask for number. `The number of failures is unknown. So, need to rephrase it.` means `the log message's statement is not guaranteed and need to rephrase it to something that will be true in any case`.